extensions/import/src/controllers/mainController.ts

```diff
@@ -25,15 +25,15 @@ export default class MainController extends ControllerBase {
 	public deactivate(): void {
 	}
 
-	public activate(): Promise<boolean> {
-		const outputChannel = vscode.window.createOutputChannel(constants.serviceName);
-		new ServiceClient(outputChannel).startService(this._context);
-
-		managerInstance.onRegisteredApi<FlatFileProvider>(ApiType.FlatFileProvider)(provider => {
-			this.initializeFlatFileProvider(provider);
+	public async activate(): Promise<boolean> {
+		return new Promise<boolean>(async (resolve) => {
+			const outputChannel = vscode.window.createOutputChannel(constants.serviceName);
+			await new ServiceClient(outputChannel).startService(this._context);
+			managerInstance.onRegisteredApi<FlatFileProvider>(ApiType.FlatFileProvider)(provider => {
```

Comment:
This callback should be set before the call to startService so we don't possibly miss the event being fired.

Reply:
Yeah I missed this

Reply:
Hi @Charles-Gagnon, should I move this piece of code into an async function and call it from here?

Reply:
What do you mean? This is already an async function. But no - just move this line to right above the `await new ServiceClient(...` line is enough